network-mux/src/Network/Mux/Bearer/AttenuatedChannel.hs

```diff
@@ -85,16 +82,22 @@ readQueueChannel QueueChannel { qcRead } =
 
 writeQueueChannel :: MonadSTM m
                   => QueueChannel m -> Message -> m Bool
-writeQueueChannel QueueChannel { qcWrite } msg =
+writeQueueChannel QueueChannel { qcWrite, qcRead } msg =
     atomically $ do
       mq <- readTVar qcWrite
+      -- Match SO_LINGER set with 0 interval, by not writing MsgClose
+      -- and closing this end without any waiting for any ack. It is
+      -- simulating a lost message, so if the receiver does not get the packet,
+      -- it will get an error the next time it tries to send a packet, closing
+      -- its end.
       case mq of
-        Nothing -> return False
-        Just q  -> writeTQueue q msg
-                >> case msg of
-                      MsgClose -> writeTVar qcWrite Nothing
-                      _        -> return ()
-                >> return True
+        Nothing -> writeTVar qcRead Nothing
+                >> return False
+        Just q  ->
+          case msg of
+            MsgClose -> writeTVar qcWrite Nothing
+            _        -> writeTQueue q msg
+          >> return True
```

Comment:
`do` notation is probably a better choice. `>>` nicely inlines with `->`, I like `$>` though some don't (e.g. Duncan).

Reply:
Addressed!